CONTRIBUTING.md

```diff
@@ -1,65 +1,189 @@
+
+        $ make tests         # Application tests
+        $./bin/unit_tests    # Unit tests
+
+	Exiv2's test system is described on more detail in Appendix [A.1](#a1-the-exiv2-test-system) (TODO)
+
+8. Push the changes to your fork on GitHub:
+
+        $ git push origin my_first_pr_branch
+
+9. Create the PR by pressing the _New pull request_ button on: `https://github.com/your-username/exiv2`
+
+
+10. Now wait for an exiv2 project member(s) to respond to your PR. Follow the discussion on your PR at [https://github.com/Exiv2/exiv2/pulls](GitHub). 
+   You may have to do some updates to your PR until it gets accepted.
 
-4. Make your change. Add tests for your change. Make the tests pass.
-5. Create the Pull Request (PR).
-    a. Check if the CI checks pass.
-    b. If your PR lives for a long time, try to avoid to press the button "Update branch" in the Pull Request view. We prefer rebasing in top of master, instead of merging master into branches.
-    c. A project member will review your PR as soon as possible.
 
-## Pull Request guidelines
-- PRs should be kept at a manageable size. Try to focus in just one goal per PR. If you find yourself doing several things that in a PR that were not expected, try to split the different tasks in different PRs.
-- Commits should always change one logical unit, so that cherry-picking & reverting is simple.
-- Commit messages should as informative and concise as possible. The first line of the commit message should be < 80 characters and describe the commit briefly. If the 80 characters are too short for a summary, consider splitting the commit. Below the short summary add one blank line and then a more detailed explanation if required.
+11. After the PR has been reviewed you must _rebase_ your repo copy since there may have been several changes to the main upstream repo 
+   (a PR can live for several days or even weeks).
 
-## Code guidelines ##
-### General ###
+	First, fetch stuff from exiv2_upstream (it does not matter in which branch you are currently in on your local copy of the repository):
+
+        $ git fetch exiv2_upstream
+
+	Then go to the master branch (the one corresponding to your fork):
+
+        $ git checkout master
+
+	Now, merge the changes in `exiv2_upstream/master` into your master branch:
+
+        $ git merge exiv2_upstream/master
+
+	At this point you can push the changes to your fork (just to be sure that you do not need to do this again):
+
+        $ git push
+
+	Now, go back to the branch:
+
+        $ git checkout my_first_pr_branch
+
+	And I rabase in top of master:
+
+        $ git rebase master
+
+	When you do a rebase the commit history is rewritten and, therefore, the next time you try to push your branch to your fork repository you will need to use the `--force` option:
+
+        $ git push --force
+
+Important: If your PR lives for a long time, then try to avoid to press the button _Update branch_ in the Pull Request view - we prefer rebasing in top of master, 
```

Comment:
This is a remainder from the original that is ambiguous, I'd reword it in the following way (and maybe it should be moved further to the start of section 11):

Important: If your PR lives for a long time, then don't press the button _Update branch_ in the Pull Request view, instead follow the steps described above, as that avoids additional merge commits.

Reply:
OK, I can move it to the start of the section.